src/buffer4/worker.rs

```diff
@@ -225,3 +225,7 @@ impl Handle {
             .unwrap_or_else(|| Closed::new().into())
     }
 }
+
+async fn recv_option<T>(x: Option<&mut mpsc::Receiver<Option<T>>>)) -> Option<T> {
```

Comment:
woops, didn't notice
```suggestion
async fn recv_option<T>(x: Option<&mut mpsc::Receiver<Option<T>>>) -> Option<T> {
```